subxt/src/blocks/extrinsic_types.rs

```diff
@@ -215,11 +215,13 @@ where
 
         let metadata = client.metadata();
 
+        let decoded_extrinsic_bytes: Vec<u8> = Decode::decode(&mut &encoded_extrinsic_bytes[..])?;
```

Comment:
doesn't this mess up the indexing after decoding it here?

Reply:
This is basically just removing the prefix of the bytes. The idea in this PR is that we keep the prefix for as long as possible in the `ChainBlockExtrinsic` and only remove it (e.g. calling `Decode::decode` when we actually need to get details out of the extrinsic bytes).